frame/transaction-payment/src/lib.rs

```diff
@@ -134,7 +136,9 @@ impl<T: Trait> Module<T> {
 #[derive(Encode, Decode, Clone, Eq, PartialEq)]
 pub struct ChargeTransactionPayment<T: Trait + Send + Sync>(#[codec(compact)] BalanceOf<T>);
 
-impl<T: Trait + Send + Sync> ChargeTransactionPayment<T> {
+impl<T: Trait + Send + Sync> ChargeTransactionPayment<T> where
+	T::Call: Dispatchable<Info=DispatchInfo>,
```

Comment:
ditto. This is going to be extremely annoying.

Reply:
We cannot avoid these because they would require adding `Dispatchable<Info=DispatchInfo>`to `frame_system::Call` and not only `Dispatchable`. This would make `()` not possible to use cause we cannot use `DispatchInfo` in our `()` impl which is in primitives.

Reply:
So basically that `Dispatchable` in primitives is preventing that.

Reply:
Maybe the only other option is to not use `()` and use a `testing::Call`. But I think that's also an annoyance. I am fine with the current state of affairs.